includes/class-wc-order-factory.php

```diff
@@ -46,7 +47,7 @@ public static function get_order( $order_id = false ) {
 		try {
 			return new $classname( $order_id );
 		} catch ( Exception $e ) {
-			wc_caught_exception( $e, __FUNCTION__, func_get_args() );
+			wc_caught_exception( $e, __FUNCTION__, $passed_args );
```

Comment:
Maybe we don't need to preserve the old PHP < 7.0 behavior and pass the original value to `wc_caught_exception()`? I don't see a reason to pass the original value as the exception will be raised when trying to instantiate the class using the current value of `$order_id`. Or am I missing something?

I also think we should pass an array to `wc_caught_exception()` as this is what the function expects and also as `func_get_args()` returns an array.

To summarize, I believe we could simply change this line to:

```
 wc_caught_exception( $e, __FUNCTION__, array( $order_id ) ); 
```

And remove the line that sets `$passed_args`.

What do you think?

Reply:
That makes a lot of sense. I'll make those changes.